plugins/storage/image/swift/src/org/apache/cloudstack/storage/datastore/driver/SwiftImageStoreDriverImpl.java

```diff
@@ -67,7 +73,28 @@ public DataStoreTO getStoreTO(DataStore store) {
 
     @Override
     public String createEntityExtractUrl(DataStore store, String installPath, ImageFormat format, DataObject dataObject) {
-        throw new UnsupportedServiceException("Extract entity url is not yet supported for Swift image store provider");
+
+        SwiftTO swiftTO = (SwiftTO)store.getTO();
+        String tempKey = UUID.randomUUID().toString();
+        boolean result = SwiftUtil.setTempKey(swiftTO, tempKey);
+
+        if (!result) {
+            s_logger.error("Unable to set Temp-Key: " + tempKey);
+            return "";
+        }
+
+        String containerName = SwiftUtil.getContainerName(dataObject.getType().toString(), dataObject.getId());
+        String objectName = installPath.split("\\/")[1];
+        // Get extract url expiration interval set in global configuration (in seconds)
+        int urlExpirationInterval = Integer.parseInt(_configDao.getValue(Config.ExtractURLExpirationInterval.toString()));
+
+        URL swiftUrl = SwiftUtil.generateTempUrl(swiftTO, containerName, objectName, tempKey, urlExpirationInterval);
+        if (swiftUrl != null) {
+            s_logger.info("Swift temp-url: " + swiftUrl.toString());
+            return swiftUrl.toString();
+        }
+
```

Comment:
Could instead add a log statement here that indicates that the extraction url could not be created for the container and object